app/models/missing_data_calculator.rb

```diff
@@ -75,10 +75,58 @@ def months_string(months, metric_name)
       end
     }.compact.sort
 
-    # date_list is a list of dates where we do have data, and we want to know what dates
-    # we don't have data
-    dummy(start_date, end_date, date_list)
+    months_missing(start_date, end_date, date_list)
   end
 
-  def dummy(_start, _end, _list); end
+  def datetime_sequence(start, stop, step)
+    dates = [start]
+    while dates.last < (stop - step)
+      dates << (dates.last + step)
+    end
+    dates
+  end
+
+  def months_missing(start_date, end_date, date_list)
```

Comment:
Can we do this with Set?  Perhaps datetime_sequence could be datetime_set and then it's just a Set#difference call if date_list was also a set.

Reply:
thanks @rossjones. Have amended.